### DISCUSSION TITLE
jupyterlite >=0.1.0b9 - OSError: [Errno 28] Invalid argument

### DISCUSSION THREAD

**Discussion by eugene-f02** [jupyterlite/jupyterlite#870] [Q&A]
Hello, I'm reaching out regarding an issue that I have been experiencing with jupyterlite ever since the 0.1.0b9 release.

As you can see in the picture below, it's from https://jupyterlite.github.io/demo/lab/index.html, the jupyterlite's demo site which is currently at jupyterlite==0.1.0b14 and jupyuterlab==3.4.8. I created a new notebook and tried to create `test.txt` file but I'm getting  OSError: [Errno 28] Invalid argument error. Note that `os.getcwd()` returns `/drive`.

<img width="1432" alt="Screen Shot 2022-11-09 at 1 36 08 PM" src="https://user-images.githubusercontent.com/57968389/200914571-03ca6e45-be8d-4dfb-a849-e3d75b4c0105.png">

Notice that If I use `/home/pyodide/test.txt` filename instead of `test.txt`, it seems to work even though the file is not visible in the file browser on the left. 
<img width="1432" alt="Screen Shot 2022-11-09 at 1 52 57 PM" src="https://user-images.githubusercontent.com/57968389/200916369-275c7a36-9fe7-4a33-84d3-48d6da175b74.png">

When I tried to build a jupyterlite instance with  jupyterla==3.3.4 and jupyterlite==0.1.0b8, here is what I discovered:

<img width="1432" alt="Screen Shot 2022-11-09 at 1 41 26 PM" src="https://user-images.githubusercontent.com/57968389/200914580-863b3acf-8b5c-4d0f-a80c-e14f5363cb0b.png">

 `os.getcwd()` returns `/home/pyodide` and 
```
with open("test.txt", "w+") as fp:
    fp.write("hi")
```
also works.

It seems that something is definitely not working right in the latest versions of jupyterlite. It might be related to services.js. Perhaps, the issue was introduced in this [pr](https://github.com/jupyterlite/jupyterlite/pull/655).

Thank you

**Comment by bollwyvl** [comment]
There are some expected differences when the serviceworker is not enabled, but at this point we pretty much need to keep it, and fix the issues, in order to keep unified kernel/client contents.

At present, you can try disabling the serviceworker (and all the features it provides) altogether in `jupyter-lite.json`:

```json
{
  "jupyter-lite-schema-version": 0,
  "jupyter-config-data": {
    "disabledExtensions": [
      "@jupyterlite/server-extension:service-worker"
    ]
  }
}
```

To meaningfully improve the situation, we'd need some more os/browser/version/mode information: for example it's pretty much known that Firefox in private browsing doesn't work with the serviceworker, but should fall back to a more reasonable state.

Additionally, there is likely more content in your browser logs (revealed with <kbd>f12</kbd>).

**Comment by jtpio** [comment]
This might indeed be related to https://github.com/jupyterlite/jupyterlite/issues/679.

> Additionally, there is likely more content in your browser logs (revealed with f12).

Yes @eugene-f02 if you could share some logs from the dev tools console that would be helpful. Also we recommend using the latest stable browsers and your first screenshot seems to indicate there is a pending update available.

**Comment by eugene-f02** [comment]
Thanks for your responses, @bollwyvl and @jtpio.

Here is some information that might be useful:

OS: macOS Monterey v12.6
Browser: Google Chrome Version 107.0.5304.110 (Official Build) (x86_64)  (I have updated the browser's version and now using the latest stable, thanks for pointing this out)
Mode: Having this issue in both incognito and default modes

<img width="1432" alt="Screen Shot 2022-11-14 at 1 01 50 PM" src="https://user-images.githubusercontent.com/57968389/201733067-89cd0f55-15cb-4d15-89ee-0fa3080a237c.png">

<img width="1432" alt="Screen Shot 2022-11-14 at 1 01 22 PM" src="https://user-images.githubusercontent.com/57968389/201732983-aa51df05-981a-4727-ba1e-d7a7765d95df.png">

Whenever I execute the cell containing

```
with open("test.txt", "w+") as f:
    f.write("hi")
```
I get a couple of failed requests as shown below:

<img width="1432" alt="Screen Shot 2022-11-14 at 1 25 34 PM" src="https://user-images.githubusercontent.com/57968389/201737367-044fee37-aacb-4969-8fda-b60edbf6c66b.png">

all of them are making requests to https://jupyterlite.readthedocs.io/en/latest/_static/api/drive 

<img width="1432" alt="Screen Shot 2022-11-14 at 1 26 05 PM" src="https://user-images.githubusercontent.com/57968389/201737455-19a5521e-63ef-44e5-b377-f418ae91d9e7.png">

and all of them return Access denied Error code 1020:

<img width="1432" alt="Screen Shot 2022-11-14 at 1 29 46 PM" src="https://user-images.githubusercontent.com/57968389/201738229-068f303e-14bc-4e43-adae-ab185cbd5bd5.png">

And here are the payloads:

<img width="1432" alt="Screen Shot 2022-11-14 at 1 26 41 PM" src="https://user-images.githubusercontent.com/57968389/201737588-78b79774-6e4d-4bc2-a061-f547fae9cf94.png">

<img width="1432" alt="Screen Shot 2022-11-14 at 1 28 16 PM" src="https://user-images.githubusercontent.com/57968389/201737882-1d588dd7-a9d6-4a9c-8b2f-dc74cced663b.png">

<img width="1432" alt="Screen Shot 2022-11-14 at 1 28 28 PM" src="https://user-images.githubusercontent.com/57968389/201737936-4211982b-c410-4670-95e9-02ec2333bcf4.png">

<img width="1432" alt="Screen Shot 2022-11-14 at 1 28 40 PM" src="https://user-images.githubusercontent.com/57968389/201738000-2dfcbc12-c7d9-4ebf-994d-96c36dbda4ae.png">

**Reply by jtpio** [reply]
Thanks @eugene-f02 for sharing the screenshots.

This is odd, I just tried with Chromium 107 (on Fedora 36) and it works fine:

![image](https://user-images.githubusercontent.com/591645/201858527-06f1cdbb-2f96-4fd6-86a2-930ca4bc146a.png)

Would you be able to manually unregister the service worker in the dev tools, or check whether it is correctly registered? Maybe that can help troubleshoot things more:

![image](https://user-images.githubusercontent.com/591645/201858857-56dcbc56-4037-4dc9-886d-8dd7802a580b.png)
